app/Http/Controllers/API/CustomerController.php

```diff
@@ -33,4 +37,31 @@ public function register(RegisterCustomerRequest $request): JsonResponse
             ->json(['message' => "The customer (#$customer->id) is created successfully."])
             ->setStatusCode(201);
     }
+
+    public  function login(LoginRequest $request):JsonResponse|LoginResource
+    {
+
+        $credentials =   $request->validated();
+        $customer = Customer::where('email', $credentials["email"])->first();
+
+        if (!$customer || ! Hash::check($credentials["password"], $customer->password)) {
+
+            return response()
+                ->json(['error' => 'The provided credentials are incorrect.'], 401);
+        }
+        $token = $customer->createToken('api_token');
+        $customer = Customer::find($customer->id);
+        $customer->last_login =$token->accessToken->created_at;
+        $customer->save();
+
+        return LoginResource::make($token);
+    }
+
+    public function  logout(Request $request):JsonResponse
+    {
+        $request->user()->currentAccessToken()->delete();
+        return response()
+            ->json(['message' => "The customer logged out successfully."]);
+
+    }
```

Comment:
Test this api with no token in header
you can create a request that just checks authorization